Lib/csv.py

```diff
@@ -409,14 +409,10 @@ def has_header(self, sample):
                 continue # skip rows that have irregular number of columns
 
             for col in list(columnTypes.keys()):
-
-                for thisType in [int, float, complex]:
-                    try:
-                        thisType(row[col])
-                        break
-                    except (ValueError, OverflowError):
-                        pass
-                else:
+                thisType = complex
+                try:
+                    thisType(row[col])
```

Comment:
Now that there’s no more loop, the loop variable seems unnecessary:

```suggestion
                try:
                    complex(row[col])
```